Description: Drop all unused extended version handling
Author: Bastian Blank <waldi@debian.org>
Date: Sun, 4 Dec 2016 12:13:49 +0100
Last-Update: 2019-08-29
Author: Noah Meyerhans <noahm@debian.org>

Index: cloud-init/tools/read-version
===================================================================
--- cloud-init.orig/tools/read-version
+++ cloud-init/tools/read-version
@@ -23,15 +23,6 @@ def tiny_p(cmd):
 
 
 def is_gitdir(path):
-    # Return boolean indicating if path is a git tree.
-    git_meta = os.path.join(path, ".git")
-    if os.path.isdir(git_meta):
-        return True
-    if os.path.exists(git_meta):
-        # in a git worktree, .git is a file with 'gitdir: x'
-        with open(git_meta, "rb") as fp:
-            if b"gitdir:" in fp.read():
-                return True
     return False
 
 
