From: Thomas Goirand <zigo@debian.org>
Date: Fri, 8 Feb 2019 14:45:04 +0100
Subject: cloud-init before chronyd

Bug-Debian: https://bugs.debian.org/910801
Forwarded: no
Last-Update: 2018-10-15
---
 systemd/cloud-init.service.tmpl | 1 +
 1 file changed, 1 insertion(+)

Index: cloud-init/systemd/cloud-init-network.service.tmpl
===================================================================
--- cloud-init.orig/systemd/cloud-init-network.service.tmpl
+++ cloud-init/systemd/cloud-init-network.service.tmpl
@@ -28,6 +28,7 @@ After=wicked.service
 After=dbus.service
 {% endif %}
 Before=network-online.target
+Before=chronyd.service
 Before=sshd-keygen.service
 Before=sshd.service
 Before=systemd-user-sessions.service
